backends/hospital-service/main.bal

```diff
@@ -118,10 +141,52 @@ service /healthcare on ln {
                 return {status: "appointment removed successfully"};
             }
         }
-        return {status: "unknown appointment ID"};
+        return {body: "unknown appointment ID"};
+    }
+
+    resource function post payments(PaymentSettlement & readonly paymentSettlement) returns Payment|http:NotFound|error {
+        lock {
+            if !appointments.hasKey(paymentSettlement.appointmentNumber) {
+                return {body: "unknown appointment ID"};        
+            }
+        }
+        PaymentSettlement {
+            patient: {name: patientName, dob}, 
+            doctor: {name: doctorName},
+            appointmentNumber
+        } = paymentSettlement;
+        int discount = let int age = check getAge(dob) in  
+                                age < 12 ? 15 : (age > 55 ? 20 : 0);
+        Doctor {fee} = check getDoctor(doctorName).ensureType();
+        final Payment & readonly payment = {
+            discounted: (100 - discount) * fee / 100,
+            doctorName,
+            patient: patientName,
+            discount,
+            appointmentNo: appointmentNumber,
+            actualFee: fee,
+            status: "settled"
+        };
+        lock {
+            payments.put(payment);
+        }
+        return payment;
     }
 
-    // todo: payments 
+    resource function get payments() returns Payment[] {
+        lock {
+            return <readonly> from Payment payment in payments select payment;
+        }
+    }
+
+    resource function get payments/payment/[string payment_id]() returns Payment|http:NotFound {
```

Comment:
`/payments/payment/<payment_id>` feels a little bit off the REST best practices. How about using `/payments/<payment_id>`?

If it is to identify this as a separate service,Let's go with the current.

Reply:
Right now, this is a 1:1 match with WSO2 backend. IMO, we can do any and all significant changes later, once we have all the tutorials.

Reply:
Ack